brili.ts

```diff
@@ -1,6 +1,72 @@
 import * as bril from "./bril-ts/bril.ts";
 import { readStdin, unreachable } from "./bril-ts/util.ts";
 
+// tiny flag parser
+const NO_WORKERS = Deno.args.includes("--no-workers");
+if (NO_WORKERS) {
+  // remove the flag so the rest of the code still sees just program args
+  Deno.args.splice(Deno.args.indexOf("--no-workers"), 1);
+}
+
+/**
+ * Opaque handle returned by `spawn` and consumed by `join`
+ */
+class ThreadHandle {
+  constructor(readonly id: number) { }
+  toString() {
+    return `<thread ${this.id}>`;
+  }
+}
+
+type ThreadRecord = {
+  promise: Promise<void>;
+  resolve: () => void;
+  reject: (e: unknown) => void;
+  used: boolean;
+};
+
+const threadTable = new Map<number, ThreadRecord>();
+const workerPorts = new Map<number, Worker>();   // tid → worker
+let nextTid = 1;
+let globalHeap: Heap<Value>;
```

Comment:
Let's avoid introducing global mutable state by moving these fields to the `State` object. That object is meant to encapsulate all the state for a running program, and things like the thread table can be considered part of that.

I am also guessing that this would eliminate the need for `globalHeap`, because we'd have access to `State.heap`.